locking down, contents and loc, updating optional fields

hannahtuttle opened the pull request.

hannahtuttle committed:
locking down, contents and loc, updating optional fields

jacobheun commented on README.md:
```diff
@@ -36,20 +41,20 @@ footnote: **Optional** - string/null
 
 * severity   : **Mandatory** - string - current expected severity - "High" | "Medium" | "Low" | "Gas" | "Refactoring" | "NonCritical" | "Disputed"
-   
+
```
While we're doing this it would be good to take a pass at all the text descriptions and have those also updated in the schema. We can probably end up just generating the readme details from the schema directly, I imagine there's generators out there already. Severity will probably be fine if we find a generator as it should clarify it must be one of the listed values.

jacobheun commented on README.md:
```diff
@@ -36,20 +41,20 @@ footnote: **Optional** - string/null
-* description: **Mandatory** - string
-   
+
+* description: **Optional** - string/null
```
For comment, footnote, and description it would be great to include something like "Accepts plain strings or valid markdown", so it's clear to folks they can style those sections.

jacobheun commented on README.md:
```diff
@@ -11,15 +11,20 @@ curl https://raw.githubusercontent.com/code-423n4/Bot-Race-JSON-Schema/v0.1.0/sc
 
-A new version of the schema will replace v0.1.0 soon (dates to be announced) and can be found [here](https://github.com/code-423n4/Bot-Race-JSON-Schema/blob/v0.2.0/schema.json).
-This version restricts the values of `severity`, making it an enum instead of an open string.
+A new version of the schema will replace v0.1.0 on Jan 1st and can be found [here](https://github.com/code-423n4/Bot-Race-JSON-Schema/blob/v0.3.0/schema.json).
```
We need to ensure we're coordinating and announcing this before we officially set Jan 1 as the date.

hannahtuttle replied:
Yes, sorry I meant to leave a note here that this a place holder date, so we can make this live once the rest of the changes we maker are ready to go live and we can add a date in here. I will add a better place holder here so it's more clear.

jacobheun commented on schema.json:
```diff
@@ -48,13 +48,15 @@
                                     "description": "Content, code snippets, @audit tags, and the file would need to be. You may choose to stack all content for an individual file here as a block, or treat it as an individual instance. If you wish to treat these as seperate instances instead of a block of instances, and you want the file to be present for each instance, please ensure that you incl ...
-                                    "type": "string"
+                                    "type": ["string", "null"],
+                                    "pattern": "^(`{3})[^`]*(\\1)$"
```
I'll pull in this latest version into the new wip lib to add some testing, but at quick glance this looks correct and should account for both opening and closing code block

jacobheun commented on schema.json:
```diff
@@ -48,13 +48,15 @@
                                     "items": {
-                                        "type": "string"
+                                        "type": "string",
+                                        "pattern": "https:\/\/github.com\/"
```
To be clear are you intending for the format to be `https://github.com/code-423n4/.../file.sol#L100`? 

The predominate usage from bots today follows the format `[100](https://github.com/code-423n4/.../file.sol#L100)`.

I'm not opposed to either but this pattern will allow both and we should land on one format. The main benefit of the first is it's the minimal version giving more flexibility when we go to render and not storing extraneous data in the database.

Either way we'll want to lock down the pattern more.

hannahtuttle replied:
I was trying to allow for both here, what I am trying to avoid is someone not adding a link. I have seen a few examples that were just the number and not a link. By locking down the pattern more are you suggestion that we add a check for a `#` and the a number, or should I be checking for `file.sol#` and then a valid number? Is there something more we should be checking for?

jacobheun replied:
Ultimately we're looking for absolute URLs that link to lines of code so we should probably be validating urls are in one of the following formats:
- `https://github.com/...#L6`
- `https://github.com/...#L12-L15`
- ~~`https://github.com/...#L12C4-L12C15`~~ (I think we probably don't need to support character linking although github technically supports it)

hannahtuttle replied:
I am looking at something like this. I still need to test that it works.

`/https:\/\/github.com\/code-423n4\/.*(.sol#L[0-9]+/i)$`

edit: this still need some though because sometime the links are `.sol#L1-L4` format at the end

hannahtuttle replied:
@jacobheun this is what I have and should work for all loc links in either format.

`https:\\/\\/github\\.com\\/[^\\/]+\\/.*\\.sol#L\\d+(-L\\d+)?\\)?$`

hannahtuttle committed:
added better styling for the read me description of the json schema. updated pattern for github link loc

hannahtuttle committed:
test rendering of pattern in read me

hannahtuttle committed:
fixing read me rendering pf pattern

hannahtuttle committed:
trying excaping teh back ticks

hannahtuttle committed:
pattern still renders incorrectly

hannahtuttle committed:
adding more details to findings params in read me

hannahtuttle committed:
updated loc pattern

jacobheun commented on README.md:
````diff
@@ -11,45 +11,119 @@ curl https://raw.githubusercontent.com/code-423n4/Bot-Race-JSON-Schema/v0.1.0/sc
 
 ```
-curl https://raw.githubusercontent.com/code-423n4/Bot-Race-JSON-Schema/v0.2.0/schema.json -o schema.json
+curl https://raw.githubusercontent.com/code-423n4/Bot-Race-JSON-Schema/v0.3.0/schema.json -o schema.json
 npx ajv-cli --spec=draft2020 -s ./schema.json -d report.json
````
Just a reminder to myself we'll want to update the examples when we have a lib released.

jacobheun commented on README.md:
```diff
@@ -11,45 +11,119 @@ curl https://raw.githubusercontent.com/code-423n4/Bot-Race-JSON-Schema/v0.1.0/sc
+
+### Category
+
+This is for making easier categorization of the issue type.
```
```suggestion
If category of the issue is clear it can be set as a plain string, otherwise if can be left empty
```

hannahtuttle committed:
Update README.md

Co-authored-by: Jacob Heun <jacobheun@gmail.com>

jacobheun commented on README.md:
```diff
@@ -11,45 +11,119 @@ curl https://raw.githubusercontent.com/code-423n4/Bot-Race-JSON-Schema/v0.1.0/sc
 
-A new version of the schema will replace v0.1.0 soon (dates to be announced) and can be found [here](https://github.com/code-423n4/Bot-Race-JSON-Schema/blob/v0.2.0/schema.json).
-This version restricts the values of `severity`, making it an enum instead of an open string.
+A new version of the schema will replace v0.1.0 on [Place Holder Date - TBD] and can be found [here](https://github.com/code-423n4/Bot-Race-JSON-Schema/blob/v0.3.0/schema.json).
```
Updating to match target release date.
```suggestion
A new version of the schema will replace v0.1.0 on **January 16th, 2024** and can be found [here](https://github.com/code-423n4/Bot-Race-JSON-Schema/blob/v0.3.0/schema.json).
```

jacobheun committed:
chore: swap out example reports with a more recent one

This was taken from the https://github.com/code-423n4/2024-01-renft audit. The only modifications that were made were to remove set loc arrays that had empty strings to empty arrays, in order to comply with the upcoming schame 0.3

jacobheun committed:
feat: update schema to latest 0.3 version

Update readme for the latest schema changes
